internal/resources/kustomization/resource_kustomization_mock_test.go

```diff
@@ -0,0 +1,581 @@
+					Source: &kustomizationclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdKustomizationRepositoryReference{
+						Name:      "someGitRepository",
+						Namespace: "tanzu-continuousdelivery-resources",
+					},
+				},
+			},
+			Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+				ParentReferences: referenceArray,
+				Description:      "resource with description",
+				Labels: map[string]string{
+					"key1": "value1",
+					"key2": "value2",
+				},
+				UID:             "cdkustomization1",
+				ResourceVersion: "v1",
+			},
+			Status: &kustomizationclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdKustomizationStatus{
+				Phase: statusmodel.NewVmwareTanzuManageV1alpha1CommonBatchPhase(statusmodel.VmwareTanzuManageV1alpha1CommonBatchPhaseAPPLIED),
+			},
+		}
+
+		getCGResponse := &kustomizationclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdKustomizationGetKustomizationResponse{
+			Kustomization: getCGModel,
+		}
+
+		getCGKustomizationEndpoint := (helper.ConstructRequestURL(https, endpoint, cgAPIVersionAndGroup, testConfig.ScopeHelperResources.ClusterGroup.Name, "namespace", apiKind, testConfig.KustomizationName)).String()
+
+		httpmock.RegisterResponder("GET", getCGKustomizationEndpoint,
+			bodyInspectingResponder(t, nil, 200, getCGResponse))
+	}
+}
+
+func (testConfig *testAcceptanceConfig) setupHTTPMocks(t *testing.T) {
+	httpmock.Activate()
+	t.Cleanup(httpmock.Deactivate)
+
+	endpoint := os.Getenv("TMC_ENDPOINT")
+
+	OrgID := os.Getenv("ORG_ID")
+
+	reference := objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference{
+		Rid: "test_rid",
+		UID: "test_uid",
+	}
+	referenceArray := make([]*objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference, 0)
+	referenceArray = append(referenceArray, &reference)
+
+	// cluster level Kustomization resorce.
+
+	postRequest, postResponse, getResponse, postContinuousDeliveryRequest, postContinuousDeliveryResponse := testConfig.getClRequest(OrgID, referenceArray)
+
+	postEndpoint := (helper.ConstructRequestURL(https, endpoint, clAPIVersionAndGroup, testConfig.ScopeHelperResources.Cluster.Name, apiSubGroup, testConfig.Namespace, apiKind)).String()
+	getKustomizationEndpoint := (helper.ConstructRequestURL(https, endpoint, clAPIVersionAndGroup, testConfig.ScopeHelperResources.Cluster.Name, apiSubGroup, testConfig.Namespace, apiKind, testConfig.KustomizationName)).String()
+	deleteEndpoint := getKustomizationEndpoint
+
+	postContinuousDeliveryEndpoint := (helper.ConstructRequestURL(https, endpoint, clAPIVersionAndGroup, testConfig.ScopeHelperResources.Cluster.Name, cdAPIKind)).String()
+
+	httpmock.RegisterResponder("POST", postContinuousDeliveryEndpoint,
+		bodyInspectingResponder(t, postContinuousDeliveryRequest, 200, postContinuousDeliveryResponse))
+
+	httpmock.RegisterResponder("POST", postEndpoint,
+		bodyInspectingResponder(t, postRequest, 200, postResponse))
+
+	httpmock.RegisterResponder("PUT", getKustomizationEndpoint,
```

Comment:
Is the request and response is same as POST?

Reply:
updated PTAL